File: gitdot-api-ts/src/endpoint/question/vote-answer.ts
import { z } from "zod";

import { VoteResource } from "../../resource";

export const VoteAnswerRequest = z.object({
  value: z.number().int(),
});
export type VoteAnswerRequest = z.infer<typeof VoteAnswerRequest>;

export const VoteAnswer = {
  path: "/repository/{owner}/{repo}/question/{number}/answer/{answer_id}/vote",
  method: "POST",
  request: VoteAnswerRequest,
  response: VoteResource,
} as const;
export type VoteAnswer = typeof VoteAnswer;

File: gitdot-api/src/endpoint/question/vote_answer.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::VoteResource};

pub struct VoteAnswer;

impl Endpoint for VoteAnswer {
    const PATH: &'static str =
        "/repository/{owner}/{repo}/question/{number}/answer/{answer_id}/vote";
    const METHOD: http::Method = http::Method::POST;

    type Request = VoteAnswerRequest;
    type Response = VoteAnswerResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct VoteAnswerRequest {
    pub value: i16,
}

pub type VoteAnswerResponse = VoteResource;
